vector/src/main/java/im/vector/activity/RoomDirectoryPickerActivity.java

```diff
@@ -210,6 +199,64 @@ public void onUnexpectedError(Exception e) {
         });
     }
 
+    /**
+     * Display a dialog to enter a custom HS url.
+     */
+    private void displayCustomDirectoryDialog() {
+        AlertDialog.Builder alert = new AlertDialog.Builder(this);
+        View dialogView = LayoutInflater.from(this).inflate(R.layout.dialog_directory_picker, null);
+        alert.setView(dialogView);
+
+        final EditText editText = (EditText) dialogView.findViewById(R.id.directory_picker_edit_text);
+
+        alert.setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
+            public void onClick(DialogInterface dialog, int whichButton) {
+                final String serverUrl = editText.getText().toString().trim();
+
+                if (!TextUtils.isEmpty(serverUrl)) {
+                    mLoadingView.setVisibility(View.VISIBLE);
+                    mSession.getEventsApiClient().getPublicRoomsCount(serverUrl, new ApiCallback<Integer>() {
+                        @Override
+                        public void onSuccess(Integer count) {
+                            Intent intent = new Intent();
+                            intent.putExtra(EXTRA_OUT_ROOM_DIRECTORY_DATA, new RoomDirectoryData(serverUrl, serverUrl, null, null, false));
+                            setResult(RESULT_OK, intent);
+                            finish();
+                        }
+
+                        private void onError(String error) {
+                            Log.e(LOG_TAG, "## onSelectDirectoryServer() failed " + error);
+                            mLoadingView.setVisibility(View.GONE);
```

Comment:
Maybe activity has been killed before we get the request response, which would cause a crash